Make Paths Windows‐Compatible

SDGGiesbrecht opened the pull request:
<!--
    Thanks for contributing to Swift Collections Benchmark!

    If this pull request adds a new feature, please add '?template=new.md'
    to the URL to switch to the appropriate template.

    Before you submit your request, please replace the paragraph
    below with the relevant details, and complete the steps in the
    checklist by placing an 'x' in each box:
    
    - [x] I've completed this task
    - [ ] This task isn't completed
-->

This resolves swift‐collections‐benchmark’s portion of apple/swift-collections#33.

### Checklist
- [x] I've read the [Contribution Guidelines](https://github.com/apple/swift-collections-benchmark#contributing-to-swift-collections-benchmark)
- [x] My contributions are licensed under the [Swift license](https://swift.org/LICENSE.txt).
- [x] I've followed the coding style of the rest of the project.
- [x] I've added tests covering my changes (if appropriate).
  - **I have included a Windows GitHub action that will catch regressions, but I can remove it again if you would prefer.**
- [ ] I've verified that my change compiles and works correctly, and does not break any existing tests.
  - **I searched for the filenames across the project and hopefully found and changed everything that pointed at them.**
  - **I do not know whether [these identifiers](https://github.com/apple/swift-collections/blob/fe745fc2e8f69b2d5e8990f82154bc5d590a703f/Documentation/Announcement-benchmarks/Library.json#L8) refer to the images or not, or if they are user‐facing. It also is not clear to me whether they are reused in generating documentation, or if they are just an archived example. Let me know what you want done with them.**
- [N/A] I've updated the documentation (if appropriate).

SDGGiesbrecht committed:
Added Windows job.

SDGGiesbrecht committed:
Renamed files.

SDGGiesbrecht committed:
Fixed image references.

lorentey commented on .github/workflows/Windows.yaml:
```diff
@@ -0,0 +1,11 @@
+    runs-on: windows-2019
+    steps:
+    - name: Check out
+      uses: actions/checkout@v1
```
Unfortunately we aren't ready to use GitHub Actions at this time. We'll need to remove this.

SDGGiesbrecht replied:
Removed.

lorentey commented on Documentation/01 Getting Started.md:
```diff
@@ -248,11 +248,11 @@ With these arguments, the result is an interactive Markdown file that links to s
 >   <details>
 >     <summary>01: Deque: random access lookups</summary>
->     <img src="./Assets/sample-results/01%20Deque:%20random%20access%20lookups.png">
+>     <img src="./Assets/sample-results/01%20Deque_%20random%20access%20lookups.png">
```
This and the renamings below ought to be consistent with https://github.com/apple/swift-collections/pull/34.

```suggestion
>     <img src="./Assets/sample-results/01%20Deque%20-%20random%20access%20lookups.png">
```

SDGGiesbrecht replied:
Switched.

lorentey commented on Documentation/01 Getting Started.md:
```diff
@@ -248,11 +248,11 @@ With these arguments, the result is an interactive Markdown file that links to s
 >   <details>
 >     <summary>02: Deque: prepending individual integers</summary>
->     <img src="./Assets/sample-results/02%20Deque:%20prepending%20individual%20integers.png">
+>     <img src="./Assets/sample-results/02%20Deque_%20prepending%20individual%20integers.png">
```
```suggestion
>     <img src="./Assets/sample-results/02%20Deque%20-%20prepending%20individual%20integers.png">
```

SDGGiesbrecht replied:
Switched.

lorentey reviewed (changes requested):
These file names were generated by the tool itself, based on the [corresponding library definition](https://github.com/apple/swift-collections/blob/main/Documentation/Announcement-benchmarks/Library.json), which has been updated to use ` - ` instead of `:`, and we should use the same names here.

To prevent similar issues when people try to *run* benchmarks on Windows, it would be a good idea to make sure the tool filters out invalid characters from the filenames it generates. The filename sanitizer code is [currently really dumb](https://github.com/apple/swift-collections-benchmark/blob/main/Sources/CollectionsBenchmark/BenchmarkCLI/BenchmarkCLI%2BLibrary%2BRender.swift#L110-L111), and it's not even factored out. I like the idea of systematically replacing `/`, `\\`, `:`, `>`, `\0` etc with underscores, depending on the current system.

(It needs to be possible to use colons and angle brackets in chart titles, even if the underlying filesystem forbids their use.)

SDGGiesbrecht commented:
> These file names were generated by the tool itself, based on the `corresponding library definition`, which has been updated to use `-` instead of `:`, and we should use the same names here.

L.O.L. I was trying to work on the dependency graph from the bottom up (swift-collections-benchmarks before swift-collections), but I guess I was holding it backwards. With the original definitions changed, will all the derived stuff in this repository be automatically fixed by the next run? If so, then I recommend just doing that and scrapping this PR. This was done by hand not realizing root definitions existed elsewhere.

lorentey commented:
Oh, these won't be automatically regenerated -- synchronizing the filenames will just make things easier if these examples ever need to be recreated.

SDGGiesbrecht committed:
Removed GitHub action.

SDGGiesbrecht committed:
Switched sanitization scheme.

SDGGiesbrecht commented:
I addressed the feedback about existing changes.

> To prevent similar issues when people try to run benchmarks on Windows, it would be a good idea to make sure the tool filters out invalid characters from the filenames it generates.

I would rather leave that for someone more familiar with the Windows than I am. It does not appear that simple: https://stackoverflow.com/a/1976050/5280938

lorentey reviewed (approved):
Looks good! Thank you 😊

lorentey commented:
@swift-ci test

lorentey commented:
I submitted #10 to fix generated filenames on Windows.
